### ISSUE TITLE
Populating parameter using Depends() ignores pydantic model config

### ISSUE THREAD

**Issue by dmig-alarstudios** [tiangolo/fastapi#1974]
I use `Depends()` to populate model from query parameters. Recently noticed, that dependency resolving somehow ignores pydantic model configuration. According to #1474, it may be the same issue.

### Environment

* OS: linux
* FastAPI Version [e.g. 0.3.0]: 0.61
* Python version: 3.8.2

### Testcase
```python3
from pydantic import BaseModel, Field
from fastapi import Depends, FastAPI

app = FastAPI()

class TestParams(BaseModel):
    class Config:
        allow_population_by_field_name = True

    param: bool = Field(False, alias='alias')


@app.get('/')
async def main(params: TestParams = Depends()):
    return params.dict()
```

# Expected results
```shell
$ curl "http://127.0.0.1:8000/?param=true"
{"param":true}

$ curl "http://127.0.0.1:8000/?alias=true"
{"param":true}
```

# Actual results
```shell
$ curl "http://127.0.0.1:8000/?param=true"
{"param":false}

$ curl "http://127.0.0.1:8000/?alias=true"
{"param":true}
```

**Comment by SirTelemak** [comment]
Somehow `name` param in request validation is equal to alias. I'll try to make PR with fix today

**Comment by dmig-alarstudios** [comment]
Yeah, the problem is there by design. I believe the same issue would be with `Path`/`Body` model instantiation:
- FastAPI shouldn't pick query string parameters by names -- proper way would be to pass all of them to pydantic model
- that would also solve another minor issue: using `exclude_unset=True` on models `.dict()`/`.json()` has no effect now, because all model fields are being passed by FastAPI

**Comment by jrversteegh** [comment]
@SirTelemak 
> Somehow name param in request validation is equal to alias.

That's by design? `name` is the name in python, `alias` is the name in the API, whether json body or query string. At least, that is how I understand it.

> so there is no info about real name.

Not true, I believe. If you have:
```python
async def f(p: int = Query(..., alias='q')):
    pass
```
then:
```python
inspect.signature(f).parameters['p'].default.alias == 'q'
```

**Comment by SirTelemak** [comment]
@jrversteegh you right that alias and name here is different things but FastAPI (if I remember correctly this case) redefines initial model and creates duplicate where name equal to alias (if exists) or name. And that why i said "there is no info about real name." since we creating new model that knows nothing about initial one

**Comment by jrversteegh** [comment]
@SirTelemak 
> redefines initial model and creates duplicate where name equal to alias (if exists) or name

I highly doubt that since I'm able to use aliases that aren't valid python variable names like in
```python
async def f(page_size: int = Query(..., alias='page[size]')):
    pass
```
